packages/kit/src/runtime/client/client.js

```diff
@@ -63,6 +63,62 @@ function update_scroll_positions(index) {
 	scroll_positions[index] = scroll_state();
 }
 
+/** @param {Element} element */
+function is_scrollable(element) {
+	const style = getComputedStyle(element);
+	return style.display !== 'contents' &&
+		(style.overflowY === 'auto' || style.overflowY === 'scroll') &&
+		element.scrollHeight > element.clientHeight
+		? element
+		: null;
+}
+
+const SCROLLABLE_AREA_ATTRIBUTE = 'data-sveltekit-main-scrollable';
+
+function get_scrollable_area() {
+	const html_is_scrollable = getComputedStyle(document.documentElement).overflowY !== 'hidden';
+	const scrollable_area = html_is_scrollable
+		? document.documentElement
+		: is_scrollable(document.body);
+
+	const custom_scrollable_areas = document.querySelectorAll(`*[${SCROLLABLE_AREA_ATTRIBUTE}]`);
```

Comment:
probably better to only do `querySelector` and move the `..All` inside the DEV check